diff --git a/src/graphics/text.c b/src/graphics/text.c
index f740cd69..8f605ab4 100644
--- a/src/graphics/text.c
+++ b/src/graphics/text.c
@@ -389,16 +389,15 @@ int text_draw_multiline(const uint8_t *str, int x_offset, int y_offset, int box_
         }
         int current_width = 0;
         int line_index = 0;
-        while (has_more_characters) {
+        while (has_more_characters && current_width < box_width) {
             int word_num_chars;
             int word_width = get_word_width(str, font, &word_num_chars);
-            if (current_width + word_width >= box_width) {
+            current_width += word_width;
+            if (current_width >= box_width) {
                 if (current_width == 0) {
                     has_more_characters = 0;
                 }
-                break;
             } else {
-                current_width += word_width;
                 for (int i = 0; i < word_num_chars; i++) {
                     if (line_index == 0 && *str <= ' ') {
                         str++; // skip whitespace at start of line
@@ -436,13 +435,13 @@ int text_measure_multiline(const uint8_t *str, int box_width, font_t font, int *
         while (has_more_characters) {
             int word_num_chars;
             int word_width = get_word_width(str, font, &word_num_chars);
-            if (current_width + word_width >= box_width) {
+            current_width += word_width;
+            if (current_width >= box_width) {
                 if (current_width == 0) {
                     has_more_characters = 0;
                 }
                 break;
             } else {
-                current_width += word_width;
                 str += word_num_chars;
                 if (!*str) {
                     has_more_characters = 0;
